lib/HLSL/DxilValueCache.cpp

```diff
@@ -269,6 +310,15 @@ Value *DxilValueCache::WeakValueMap::GetSentinel(LLVMContext &Ctx) {
   return Sentinel.get();
 }
 
+void DxilValueCache::WeakValueMap::Clear() {
```

Comment:
Rename because it doesn't clear the value map, and maybe we should remove the sentinel values from the map?  Also, sentinel values could be `(Value*)-1`.